rsfbclient-core/src/connection.rs

```diff
@@ -32,7 +32,7 @@ pub trait FirebirdClient: Send {
     /// Arguments to instantiate the client
     type Args: Send + Sync + Clone;
 
-    fn new(args: Self::Args) -> Result<Self, FbError>
+    fn new(charset: Charset, args: Self::Args) -> Result<Self, FbError>
```

Comment:
I don't remember that is possible to retrieve/switch automatically the `dialect` and `charset` from the defaults in the connecting database.

Doing this could be handy for new users that don't have this concepts well understood.

Maybe having a variant/alternative `connect()` for this case would be good.

Of course any query that should be executed on connection time can hurt applications that don't use connection pooling and don't have long lived connections. This will be common in applications like little websites and microservice. For this case have the option of setting `dialect` and `charset` would optimize.

Reply:
Don't worry, the users can't access this method. We only use this internally.